cumulus/parachains/runtimes/assets/asset-hub-kusama/src/lib.rs

```diff
@@ -76,7 +76,7 @@ use parachains_common::{
 	NORMAL_DISPATCH_RATIO, SLOT_DURATION,
 };
 use sp_runtime::RuntimeDebug;
-use xcm::opaque::v3::MultiLocation;
+use xcm::latest::prelude::Location;
```

Comment:
@franciscoaguirre 
If you have some time, could you, please, check this [issue](https://github.com/paritytech/polkadot-sdk/issues/1130)? There are some thoughts/concerns/questions about using `(Multi)Location` vs `Versioned(MultiLocation)` and about compatibility vs migrations, where you also could bring some light :) (I know we discussed something similar in xcm channel long time ago, but that time I didnt have those concrete examples),

I think we should be careful with changing this import (please, check mentioned issue above).
E.g. are we sure that encoding `xcm::latest::prelude::Location` is 100%-backwards compatible with `xcm::opaque::v3::MultiLocation`? Do we have any compatibility test between `v4` vs `v3`? Do we need one? Maybe this works for `v4` -> `v3`, does it? But what about future xcm updates?

My concern here is that if encoding for `latest::(Multi)Location` is not backwards compatible, we could have problems because `(Multi)Location` is used as storage key. How to make sure and detect that we need to add migrations or whatever?

Reply:
Anything used in a storage item has to be a `VersionedLocation`, otherwise version changes will break decoding.
So we should migrate the storage items mentioned in the issue to use the correct type.
If we do that, these storage items will always be decodable (as long as the XCM version is not **too** old, but even then it can be looked up and decoded if needed), which means it doesn't really matter if v4 is backwards compatible with v3.

Aside from that, v4 is easily converted to v3 and vice-versa, with `try_into`, which will most likely never fail.
The idea of a versioned format is it will never be fully backwards compatible, but that's what the conversion functions and `Versioned*` types are for.